File: dbpro/src/main/java/kr/ac/sejong/feat/actor/MovieActorDAO.java
package kr.ac.sejong.feat.actor;

import kr.ac.sejong.config.JDBCManager;

import java.util.List;

public class MovieActorDAO {

	private MovieActorDAO() {
	}

	private static MovieActorDAO instance;

	public static MovieActorDAO getInstance() {
		if (instance == null) {
			instance = new MovieActorDAO();
		}
		return instance;
	}

	public List<MovieActorDTO> selectMovieActors() {
		final String query = "select " +
			"ACTOR_CODE, " +
			"MOVIE_CODE, " +
			"ACTOR_ROLE " +
			"	from " +
			"MOVIE_ACTOR";

		return JDBCManager
			.getInstance()
			.queryForList(query, MovieActorDTO.class);
	}
}

File: dbpro/src/main/java/kr/ac/sejong/feat/actor/MovieActorDAO.kt
package kr.ac.sejong.feat.actor

import kr.ac.sejong.config.JDBCManager

class MovieActorDAO private constructor() {
    companion object {
        @JvmStatic
        var instance: MovieActorDAO? = null
            get() {
                if (field == null) {
                    field = MovieActorDAO()
                }
                return field
            }
            private set
    }

    fun selectMovieActors(): List<MovieActorDTO> {
        val query = """
            SELECT
                ACTOR_CODE,
                MOVIE_CODE,
                ACTOR_ROLE
            FROM
                MOVIE_ACTOR
        """

        return JDBCManager
            .getInstance()
            .queryForList(query, MovieActorDTO::class.java)
    }
}